reconcile/utils/mr/auto_promoter.py

```diff
@@ -34,6 +44,47 @@ def title(self):
         return (f'[{self.name}] openshift-saas-deploy automated '
                 f'promotion {digest}')
 
+    @staticmethod
+    def init_promotion_data(channel: str,
+                            promotion: Mapping[str, Any]) -> Dict[str, Any]:
+        psc = ParentSaasConfigPromotion(
+            parent_saas=promotion["saas_file"],
+            target_config_hash=promotion[TARGET_CONFIG_HASH]
+        )
+        return {
+            "channel": channel,
+            "data": [
+                asdict(psc)
+            ]
+        }
+
+    @staticmethod
+    def process_promotion(promotion,
+                          target_promotion,
+                          target_channels):
+
+        # Existent subscribe data channel data
+        promotion_data = {v["channel"]: v["data"]
+                          for v in target_promotion.get('promotion_data', [])
+                          if v["channel"] in target_channels}
+
+        for channel in target_channels:
+            modified = False
+            channel_data = promotion_data.get(channel)
+            if channel_data is None:
+                channel_data = \
```

Comment:
i might miss or misunderstand something here. shouldn't the channel_data be applied back to the target_promotion somehow?

Reply:
oh my, you are completely right. I fixed it and extend the tests to cover this